src/nix.rs

```diff
@@ -37,3 +37,33 @@ pub fn isatty(fd: i32) -> bool {
     // No place we currently call it really cares about the difference.
     return unsafe { libc::isatty(fd) } == 1;
 }
+
+/// An enumeration of supported libc rusage types used by [`getrusage()`].
+/// NB: RUSAGE_THREAD is not supported on macOS.
+pub enum RUsage {
+    RSelf, // "Self" is a reserved keyword
+    RChildren,
+}
+
+/// A safe wrapper around `libc::getrusage()`
+pub fn getrusage(resource: RUsage) -> libc::rusage {
+    let mut rusage = std::mem::MaybeUninit::uninit();
+    let result = unsafe {
+        libc::getrusage(
+            match resource {
+                RUsage::RSelf => libc::RUSAGE_SELF,
+                RUsage::RChildren => libc::RUSAGE_CHILDREN,
+            },
+            rusage.as_mut_ptr(),
+        )
+    };
+
+    // getrusage(2) says the syscall can only fail if the dest address is invalid (EFAULT) or if the
```

Comment:
The old code invoked `perror` if `getrusage` failed - please restore that.

Reply:
Isn't it impossible for `getrusage` to fail in this case? [The FreeBSD man page](https://man.freebsd.org/cgi/man.cgi?query=getrusage#end) says it fails if:
- 'The who [resource] argument is not a valid value'
  - This is impossible since we have the `RUsage` enum
- 'The address specified	by the rusage argument is not in a valid part of the process address space'
  - This shouldn't(?) be possible because we're using `MaybeUninit`

I think the best behaviour would probably be to panic if it doesn't return 0?

Reply:
> also MaybeUninit is now unused in fish.rs and can be removed

Whoops, forgot :^)

Reply:
Both panic and perror would work in practice but if we can recover safely (from a libc bug or new error), that seems better than panicking. We have our own `perror` wrapper, no need to use `libc::perrror`.

Reply:
Sure, would it be a good idea to also have a `try_getrusage` which returns `Result`?

Reply:
> Sure, would it be a good idea to also have a try_getrusage which returns Result?

there's no use for this at the moment because no caller cares about that